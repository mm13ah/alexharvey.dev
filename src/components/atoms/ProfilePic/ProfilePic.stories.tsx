import React from 'react';
import { StoryFn, Meta } from '@storybook/react';

import { ProfilePic } from './ProfilePic';

export default {
  title: 'Atoms/ProfilePic',
  component: ProfilePic,
  args: {},
} as Meta<typeof ProfilePic>;

const Template: StoryFn<typeof ProfilePic> = () => <ProfilePic />;

export const Primary = Template.bind({});
Primary.storyName = 'ProfilePic';
