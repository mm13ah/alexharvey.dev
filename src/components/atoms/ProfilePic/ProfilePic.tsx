import Image from 'next/image';
import React from 'react';
import Pic from '../../../../public/profile.png';

export const ProfilePic = () => (
  <Image
    alt="me"
    src={Pic}
    width={200}
    height={200}
    className="rounded-full my-3"
  />
);
