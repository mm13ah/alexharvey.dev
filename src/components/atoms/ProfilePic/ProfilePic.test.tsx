import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ProfilePic } from './ProfilePic';

describe('ProfilePic', () => {
  it('should render', () => {
    render(<ProfilePic />);
  });
  it('should match a snapshot', () => {
    const tree = renderer.create(<ProfilePic />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
