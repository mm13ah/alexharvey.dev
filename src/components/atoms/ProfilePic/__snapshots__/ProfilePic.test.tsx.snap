// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePic should match a snapshot 1`] = `
<img
  alt="me"
  className="rounded-full"
  data-nimg="1"
  decoding="async"
  height={200}
  loading="lazy"
  onError={[Function]}
  onLoad={[Function]}
  src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
  srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
  style={
    Object {
      "color": "transparent",
    }
  }
  width={200}
/>
`;
